src/dpl/src/Place.cpp

```diff
@@ -859,6 +859,40 @@ bool Opendp::checkPixels(const Cell* cell,
         }
       }
     }
+    if (disallow_one_site_gaps_) {
+      // here we need to check for abutting first, if there is an abut cell
+      // then we continue as there is nothing wrong with it
+      // if there is no abut cell, we will then check cells at 1+ distances
+      // we only need to loop on all 4 sides only
+      int x_begin = max(0, x - 1);
+      int y_begin = max(0, y - 1);
+      // inclusive search, so we don't add 1 to the end
+      int x_finish = min(x_end, row_site_count_ - 1);
+      int y_finish = min(y_end, row_count_ - 1);
+
+      // loop over left and right sides
```

Comment:
You are looping over the interior of the cell as well.  You only need to check the left and right edges (x-1, x_end) and not the points in-between.